### PR TITLE
drivers: display: gc9x01: convert to MIPI DBI API

### PR DESCRIPTION
**Pull request by danieldegrasse**
Convert galaxycore GC9X01 to MIPI DBI API. In tree boards and tests
using this display have also had their devicetrees updated to use the
new MIPI DBI SPI emulated device.

### CODE REVIEW

**Comment by danieldegrasse** [comment]
@nicogou, would you be willing to test this change? I don't have hardware to verify it, but I'd like to move this display to the new MIPI DBI API

**Comment by kartben** [comment]
@joelguittet might also be able to test on `esp32s3_touch_lcd_1_28_esp32s3` which he contributed

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay [hunk 1]
```diff
@@ -27,6 +27,26 @@
...
+
+	xiao_round_display_mipi_dbi {
+		compatible = "zephyr,mipi-dbi-spi";
+		spi-dev = <&xiao_spi>;
+		dc-gpios = <&xiao_d 3 GPIO_ACTIVE_HIGH>;
+		write-only;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		gc9a01_xiao_round_display: gc9a01@0 {
```
Comment (joelguittet) [inline boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:39]: Why moving this in the device tree ? This is more difficult to review and identify the changes ....

File: tests/drivers/build_all/display/app.overlay [hunk 2]
```diff
@@ -81,34 +91,26 @@
...
-				cmd-data-gpios = <&test_gpio 1 0>;
-				reset-gpios = <&test_gpio 2 0>;
-				pixel-format = <16>;
-
-				width = <240>;
-				height = <240>;
-			};
 
-			test_led_strip_0: lpd8806@2 {
+			test_led_strip_0: lpd8806@0 {
```
Comment (joelguittet) [inline tests/drivers/build_all/display/app.overlay:101]: Why this change ?

File: boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay (hunk 1 above)
Comment (danieldegrasse) [reply boards/shields/seeed_xiao_round_display/seeed_xiao_round_display.overlay:39]: This display uses a 4 wire SPI interface for commands and data, which means it can be supported by the MIPI DBI API. For more information on this API, you can look at the initial RFC: #64587. The main idea of the API is to better support display hardware that explicitly accelerates SPI 3/4 wire mode displays. This change will allow gc9x01 displays to operate with LCD hardware using the MIPI DBI API, as well as existing SPI controllers

For cases where a SPI peripheral is used to implement the 3/4 wire mode support, we need to use an emulated MIPI DBI driver that uses the SPI API. This driver is implemented by the `zephyr,mipi-dbi-spi` compatible, so we need to declare the display as a child of this driver's node.

Hope this helps explain the rationale for this change. Would you be willing to test it?

**Comment by danieldegrasse** [comment]
> Yes I can test on the Waveshare board too.
Just for the context please, what is the benefit of the MIPI DBI API compared to the spi interface ?
I just understand it is more complicated in the declaration of the device tree...

Thanks for testing it! I added a comment inline- I hope this helps explain the reasoning behind the MIPI DBI API

File: tests/drivers/build_all/display/app.overlay (hunk 2 above)
Comment (joelguittet) [reply tests/drivers/build_all/display/app.overlay:101]: I guess a separated commit will be better ?

Comment (danieldegrasse) [reply tests/drivers/build_all/display/app.overlay:101]: Sure, I'll split this into another commit

**Review by jfischer-no — DISMISSED** [review]
LGTM

**Comment by danieldegrasse** [comment]
Rebased to clear conflicts (this likely won't be the last time, #73812 and #73946 also touch the display build all test)

**Review by jfischer-no — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by kartben** [comment]
> Rebased to clear conflicts (this likely won't be the last time

Correct :)

**Comment by danieldegrasse** [comment]
> > Rebased to clear conflicts (this likely won't be the last time
> 
> Correct :)

Rebased again- #73812 and this both touch the same build all test, so if one merges I'll update the other.

**Review by dleach02 — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
Final rebase- @jfischer-no if you have a chance to refresh +1 it would be appreciated (but I know it is late, so no worries!)

**Review by jfischer-no — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by decsny — APPROVED** [review]
